spec/index.md

```diff
@@ -2447,6 +2447,8 @@ The encoding of the signature depends on the key ID's `algorithm`:
 
 This call requires that a Schnorr key with ID `key_id` was generated by the IC and the signing functionality for that key was enabled. Otherwise, the call is is rejected.
 
+This call accepts an optional auxiliary parameter `aux`. The auxiliary parameter type `schnorr_aux` is a variant. The only currently supported variant is `bip341` which allows passing a Merkle tree root hash, which is required to implement Taproot signatures as defined in [BIP341](https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki). The `bip341` variant is anly allowed for `bip340secp256k1` signatures, and if provided the `merkle_root_hash` blob must be exactly 32 bytes long, and should be generated in accordance with BIP341's specification for `taproot_output_script`. If no auxiliary parameter is provided, then `bip340secp256k1` signatures are generated in accordance with BIP340.
```

Comment:
```suggestion
This call accepts an optional auxiliary parameter `aux`. The auxiliary parameter type `schnorr_aux` is a variant. The only currently supported variant is `bip341` which allows passing a Merkle tree root hash, which is required to implement Taproot signatures as defined in [BIP341](https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki). The `bip341` variant is only allowed for `bip340secp256k1` signatures, and if provided the `merkle_root_hash` blob must be exactly 32 bytes long, and should be generated in accordance with BIP341's specification for `taproot_output_script`. If no auxiliary parameter is provided, then `bip340secp256k1` signatures are generated in accordance with BIP340.
```

Reply:
It also may be better to reference `taproot_sign_key` instead of `taproot_output_script` because we are tweaking the secret key, although the procedure is essentially the same. It may also be reasonable to say that `merkle_root_hash` corresponds to `h` in `taproot_sign_key` of BIP341.

Reply:
I think in that sentence `taproot_output_script` is the correct reference, as it is where the Merkle root is computed. However, I agree with the general intention of the comment to be a bit more explicit about the tweaking. In particular I would suggest that we clarify which key is used to sign if we pass or not the auxiliary output. 

> It may also be reasonable to say that merkle_root_hash corresponds to h in taproot_sign_key of BIP341.

+ 1